dask_sql/input_utils/hive.py

```diff
@@ -131,24 +131,32 @@ def wrapped_read_function(location, column_information, **kwargs):
                     _,
                 ) = parsed
 
+                partition_column_information = {
+                    col: sql_to_python_type(col_type.upper())
+                    for col, col_type in partition_column_information.items()
+                }
+
                 location = partition_table_information["Location"]
                 table = wrapped_read_function(
                     location, partition_column_information, **kwargs
                 )
 
-                # Now add the additional partition columns
-                partition_values = ast.literal_eval(
-                    partition_table_information["Partition Value"]
-                )
+                partition_values = partition_table_information["Partition Value"]
+                partition_values = partition_values[1 : len(partition_values) - 1]
```

Comment:
**bug (blocking)**: Unless `len(partition_values)` is 1 (will it always?) then this will return a list object and the subsequent `.split(',')` call will fail.